types/index.d.ts

```diff
@@ -0,0 +1,65 @@
+declare module 'svelte-routing' {
+  import 'svelte2tsx/svelte-jsx';
+  import { SvelteComponent, SvelteComponentTyped } from 'svelte';
+  
+  interface RouterProps {
+    basepath?: string;
+    url?: string;
+  }
+
+  interface RouteProps {
+    path?: string;
+    component?: SvelteComponent;
+    [additionalProp: string]: unknown;
+  }
+
+  interface RouteSlots {
+    default: {
+      location: RouteLocation;
+      params: RouteParams;
+    };
+  }
+
+  type State = {
+    [k in string | number]: unknown;
+  }
+
+  interface RouteLocation {
+    pathname: string;
+    search: string;
+    hash?: string;
+    state: State;
+  }
+
+  interface RouteParams {
+    [param: string]: string;
+  }
+
+  interface LinkProps {
+    to: string;
+    replace?: boolean;
+    state?: State;
+    getProps?: (linkParams: GetPropsParams) => Record<string, any>;
+  }
+
+  interface GetPropsParams {
+    location: RouteLocation;
+    href: string;
+    isPartiallyCurrent: boolean;
+    isCurrent: boolean;
+  }
+
+	export const Router: SvelteComponentTyped<RouterProps>;
```

Comment:
Sorry to be nitpicky, but there is some mixed whitespace here.

Reply:
That's not a problem, it's always good to keep things consistent 😄

Reply:
That's not a problem, it's always good to keep things consistent